File: backend/src/routes/ai.ts
import express from 'express';
import { body } from 'express-validator';
import authMiddleware from '../middlewares/auth.js';
import validateRequest from '../middlewares/validateRequest.js';
import { supabase } from '../services/supabaseClient.js';

const router = express.Router();
router.use(authMiddleware);

router.post(
  '/query',
  [body('prompt').isString().notEmpty().withMessage('Prompt is required.')],
  validateRequest,
  async (req, res, next) => {
    try {
      const userId = req.user!.id;
      const { prompt } = req.body;
      const response = `AI analysis ready: based on this prompt, your expenses are weighted heavily toward recurring categories; adjust your budget and savings recommendations accordingly.`;
      const { error } = await supabase.from('ai_conversations').insert([{ user_id: userId, prompt, response, category: 'financial_insight' }]);
      if (error) return res.status(400).json({ error: error.message });
      res.json({ prompt, response });
    } catch (err) {
      next(err);
    }
  }
);

export default router;

File: backend/routers/ai.py
from fastapi import APIRouter, Depends, HTTPException
import uuid
from ..schemas import User, AIQuery
from ..dependencies import get_current_user
from ..mock_db import mock_storage

router = APIRouter()

@router.post("/query")
def query_ai(req: AIQuery, user: User = Depends(get_current_user)):
    if not req.prompt:
        raise HTTPException(status_code=400, detail="Prompt is required.")
        
    response_text = "AI analysis ready: based on this prompt, your expenses are weighted heavily toward recurring categories; adjust your budget and savings recommendations accordingly."
    
    conv = {
        "id": str(uuid.uuid4()),
        "user_id": user.id,
        "prompt": req.prompt,
        "response": response_text,
        "category": "financial_insight"
    }
    mock_storage["ai_conversations"].append(conv)
    
    return {"prompt": req.prompt, "response": response_text}
